pip/cmdoptions.py

```diff
@@ -35,6 +35,24 @@ def resolve_wheel_no_use_binary(options):
         fmt_ctl_no_use_wheel(control)
 
 
+def check_install_build_global(options, check_options=None):
+    """Disable wheels if per-setup.py call options are set.
+
+    :param options: The OptionParser options to update.
+    :param check_options: The options to check, if not supplied defaults to
+        options.
+    """
+    if check_options is None:
+        check_options = options
+
+    def getname(n):
+        return getattr(check_options, n, None)
+    names = ["build_options", "global_options", "install_options"]
+    if any(map(getname, names)):
+        control = options.format_control
+        fmt_ctl_no_use_wheel(control)
```

Comment:
This will trigger a deprecation warning about using ``--no-use-wheel`` right? I think it shouldn't do that in this case since the user isn't passing ``--no-use-wheel``, we're just silently disabling them.

It might make sense to issue a logger.warning through that we're ignoring all wheels due to the use of ``--build-option``, ``--global-option``, or ``--install-option``.